game/scripts/vscripts/units/swiper.lua

```diff
@@ -0,0 +1,213 @@
+
+	thisEntity:SetContextThink( "SwiperBossThink", SwiperBossThink, 1 )
+end
+
+function SwiperBossThink()
+	if GameRules:IsGamePaused() == true or GameRules:State_Get() == DOTA_GAMERULES_STATE_POST_GAME or thisEntity:IsAlive() == false then
+		return 1
+	end
+
+	-- if not thisEntity:IsIdle() then
+	-- 	return 1.0
+	-- end
+
+	if not thisEntity.bInitialized then
+		thisEntity.vInitialSpawnPos = thisEntity:GetOrigin()
+		thisEntity.bInitialized = true
+	end
+
+	local enemies = FindUnitsInRadius(
+		thisEntity:GetTeamNumber(),
+		thisEntity:GetOrigin(), nil,
+		1000,
+		DOTA_UNIT_TARGET_TEAM_ENEMY,
+		DOTA_UNIT_TARGET_HERO,
+		DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES + DOTA_UNIT_TARGET_FLAG_FOW_VISIBLE,
+		FIND_CLOSEST,
+		false
+	)
+
+	if #enemies == 0 then
+		return RetreatHome()
+	end
+
+	local canUseRush = true
+	local canUseTeleport = false
+
+	if thisEntity:GetHealth() / thisEntity:GetMaxHealth() > 0.75 then -- phase 1
+		canUseRush = false
+	elseif thisEntity:GetHealth() / thisEntity:GetMaxHealth() > 0.5 then -- phase 2
+		canUseRush = true
+	else
+		canUseTeleport = true
+	end
+
+	-- canUseRush = true
+
+	-- Swipe
+
+	local swipeRange = thisEntity.hFrontswipeAbility:GetCastRange(thisEntity:GetAbsOrigin(), thisEntity)
+	local frontSwipeEnemies = FindUnitsInCone(
+		thisEntity:GetAbsOrigin(),
+		thisEntity:GetForwardVector(),
+		swipeRange,
+		swipeRange*2,
+		thisEntity:GetTeamNumber(),
+		DOTA_UNIT_TARGET_TEAM_ENEMY,
+		DOTA_UNIT_TARGET_ALL,
+		DOTA_UNIT_TARGET_FLAG_NONE,
+		FIND_CLOSEST)
+
+	if thisEntity.hFrontswipeAbility:IsCooldownReady() and #frontSwipeEnemies > 0 then
+		return CastFrontswipe()
+	end
+
+	-- Thrust
+
+	local thrustRange = thisEntity.hThrustAbility:GetSpecialValueFor("range")
+	local thrustWidth = thisEntity.hThrustAbility:GetSpecialValueFor("width")
+	local thrustMinRange = thisEntity.hThrustAbility:GetSpecialValueFor("target_min_range")
+	local thrustTarget
+	local thrustCount = 0
+	for k,v in pairs(enemies) do
+		local d = (v:GetAbsOrigin() - thisEntity:GetAbsOrigin()):Length2D()
+		if d < thrustRange and d > thrustMinRange then
+			local thrustEnemies = FindUnitsInLine(
+				thisEntity:GetTeamNumber(),
+				thisEntity:GetAbsOrigin(),
+				v:GetAbsOrigin(),
+				nil,
+				thrustWidth,
+				DOTA_UNIT_TARGET_TEAM_ENEMY,
+				DOTA_UNIT_TARGET_ALL,
+				DOTA_UNIT_TARGET_FLAG_NONE)
+
+			if #thrustEnemies > thrustCount then
+				thrustCount = #thrustEnemies
+				thrustTarget = v:GetAbsOrigin()
+			end
+		end
+	end
+
+	if thisEntity.hThrustAbility:IsCooldownReady() and thrustTarget then
+		return CastThrust( thrustTarget )
+	end
+
+	-- Reaper's Rush
+
+	local reapersMinRange = thisEntity.hReapersRushAbility:GetSpecialValueFor("min_range")
+	local reapersMaxRange = thisEntity.hReapersRushAbility:GetSpecialValueFor("max_range")
+	local reapersRushRadius = thisEntity.hReapersRushAbility:GetSpecialValueFor("radius")
+	local reapersRushTarget
+	local moveToTarget
+	local reapersRushCount = 0
+	for k,v in pairs(enemies) do
```

Comment:
another expensive operation, you should only call it if the cooldown is ready, and can use the ability